### PR TITLE
Add testing instructions and contributing links

### PR DESCRIPTION
**Pull request by jisantuc**
This PR adds a link to the STAC spec Code of Conduct and a Contributing guide that includes instructions for running tests. I removed some references (like to `process.md`) since we don't have that and edited lightly, but largely these changes are copy-pasta.

The main thing I added is an example workflow for making a change to an extension at the top of CONTRIBUTING.md.

### CODE REVIEW

**Comment by jisantuc** [comment]
🤷🏻‍♂️

```
-- STAC Version: 1.0.0-rc.1
--- collection: Error downloading https://schemas.stacspec.org/v1.0.0-rc.1/collection-spec/json-schema/collection.json 
HTTP ERROR 404
--- collection-assets: Error downloading https://schemas.stacspec.org/v1.0.0-rc.1/extensions/collection-assets/json-schema/schema.json 
HTTP ERROR 404
--- item-assets: Error downloading https://schemas.stacspec.org/v1.0.0-rc.1/extensions/item-assets/json-schema/schema.json 
HTTP ERROR 404
--- https://stac-extensions.github.io/template/v1.0.0/schema.json: valid
```

**Comment by m-mohr** [comment]
Isn't that better off centrally defined in stac-overview? I feel like we put too much things into the extension repos.

**Comment by jisantuc** [comment]
I think it's good to locate a repo's instructions for locally running tests in the repo. I can see not including the full CONTRIBUTING here and just linking it like the CoC, but the testing and formatting commands should be here somewhere.

**Comment by m-mohr** [comment]
Yes, I agree. We should strip it down a bit, remove things that are only true for stac-spec and then I think we can merge.

File: README.md [hunk 1]
```diff
@@ -48,3 +48,51 @@ The following types should be used as applicable `rel` types in the
 | Type                | Description |
 | ------------------- | ----------- |
 | fancy-rel-type      | This link points to a fancy resource. |
+
+## Contributing
+
+All contributions are subject to the
+[STAC Specification Code of Conduct](https://github.com/radiantearth/stac-spec/blob/master/CODE_OF_CONDUCT.md).
+For contributions, please follow the
+[STAC specification contributing guide](https://github.com/radiantearth/stac-spec/CONTRIBUTING.md) Instructions
```
Comment (m-mohr) [inline README.md:57]: The link is broken.

Comment (jisantuc) [reply README.md:57]: It's because it's an artisanally crafted, hand-made link 🤦‍♂️

Comment (jisantuc) [reply README.md]: 2fcde6b

**Pull request merged**